oak_marketplace_emotion_recognition/main.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import cv2
+import numpy as np
+import skvideo.io
+from emotion_recognition_retail import InferenceModel, process_frame
+from modelplace_api.visualization import draw_emotion_recognition_one_frame
+from args_parser import parse_args
+from utils import EmotionGraphsCollector, overlay_image_alpha, place_class_names_and_percents, X_OFFSET, Y_OFFSET
+
+
+
+
+def build_model(model_path):
+    model = InferenceModel(model_path=model_path)
+    model.model_load()
+    model.add_cam_to_pipeline()
+    return model
+
+
+def main(args):
+    model_path = os.path.join(args.root_model_path, "checkpoint")
+    model = build_model(model_path)
+    input_width, input_height = model.get_input_shapes()
+    emotions_collector = EmotionGraphsCollector()
+    proceed = True
```

Comment:
The main function must not contain the code for the processing. It should only call functions depends on args. Please, move it to a separate function where processing will be done. Please, look, how it's done there https://github.com/opencv-ai/oak-model-samples/blob/main/oak_inference_utils/inference.py. All processing is done in process_cam